server/runtime/src/main/java/org/infinispan/server/security/RespClientCertAuthenticator.java

```diff
@@ -0,0 +1,53 @@
+package org.infinispan.server.security;
+
+import java.security.Principal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import javax.net.ssl.SSLPeerUnverifiedException;
+import javax.security.auth.Subject;
+import javax.security.sasl.SaslException;
+import javax.security.sasl.SaslServer;
+
+import org.infinispan.commons.util.concurrent.CompletableFutures;
+import org.infinispan.server.core.logging.Log;
+import org.infinispan.server.core.security.sasl.SaslAuthenticator;
+import org.infinispan.server.core.security.sasl.SubjectSaslServer;
+import org.infinispan.server.core.security.simple.SimpleSaslAuthenticator;
+
+import io.netty.channel.Channel;
+import io.netty.handler.ssl.SslHandler;
+
+public class RespClientCertAuthenticator {
+
+
+   public CompletionStage<Subject> clientCertAuth(Channel channel) throws SaslException {
+      SaslAuthenticator sap = new SimpleSaslAuthenticator();
```

Comment:
You should not be using the `SimpleSaslAuthenticator` but rather rely on the Elytron SASL authenticator which has been configured with a `TrustStoreRealm` and therefore can validate the client certificates against the trust store.